### PR TITLE
x-pack/filebeat/module/checkpoint: fix handling of R81 fields

### LINKED ISSUE #32380: [Filebeat][Checkpoint module] data stream timestamp field [@timestamp] is missing
Hi,

I'm trying to ingest CheckPoint native Syslog exports of security gateway (firewall) logs. My understanding is that integration was previously via CEF, which did not pass through sufficient detail, but that the native syslog format was merged here: [Checkpoint Syslog Filebeat module by P1llus · Pull Request #17682 · elastic/beats · GitHub](https://github.com/elastic/beats/pull/17682)

We had the following problem with CheckPoint R81 and continue to experience the same problem with the latest generally recommended version R81.10. We have configured the CheckPoint log exporter via SmartConsole, as follows:
image

Format is set as standard 'Syslog' format, which should include all the additional CheckPoint fields:
image

The problem we experiencing is that nothing is actually ingested, we receive the following error:
image

The input pipeline was automatically configured when we added the Check Point module to an Elastic Agent via Fleet. This input pipeline appears to refer to fields which Check Point don't appear to generate:
image

CheckPoint documentation for the description of fields in Check Point Logs does not include '@timestamp' or 'timestamp':
https://supportcenter.checkpoint.com/supportcenter/portal?eventSubmit_doGoviewsolutiondetails=&solutionid=sk144192

For confirmed bugs, please report:
- Version: 8.3.2
- Operating System: Debian 11 (bullseye)
- Discuss Forum URL: https://discuss.elastic.co/t/filebeat-checkpoint-module-data-stream-timestamp-field-timestamp-is-missing/309802
- Steps to Reproduce: Setup CheckPoint log ingestion using Elastic Agent and then configure CheckPoint log server to export logs via 'Syslog' format.

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

Fixes handling of numeric and duplicated fields and removes undocumented fields that appear to be new in R81.

## Why is it important?

Relates to a user issue; duplicated fields cause a pipeline failure and non-ingestion due to absence of `@timestamp`. The other changes keep the pipeline in sync with the integration package pipeline where the changes are required due to field type and documentation checks.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #32380

### CODE REVIEW

File: x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.yml [hunk 1]
```diff
@@ -434,6 +434,10 @@ processors:
       field: checkpoint.ifdir
       target_field: network.direction
       ignore_missing: true
+  - convert:
+      field: checkpoint.bytes
+      type: long
+      ignore_missing: true
   - rename:
       field: checkpoint.bytes
       target_field: network.bytes
```
Comment (r00tu53r) [inline x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.yml:443]: Can this be merged into a single step ? - https://github.com/elastic/integrations/pull/3800#discussion_r927481295

File: x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.yml [hunk 2]
```diff
@@ -471,6 +475,10 @@ processors:
         } else if (iana_number == '132') {
             ctx.network.transport = 'sctp';
         }
+  - convert:
+      field: checkpoint.packets
+      type: long
+      ignore_missing: true
   - rename:
       field: checkpoint.packets
       target_field: network.packets
```
Comment (r00tu53r) [inline x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.yml:484]: Can this be merged into a single step - https://github.com/elastic/integrations/pull/3800#discussion_r927482597

**Review by r00tu53r — APPROVED** [review]
LGTM

File: x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.yml (hunk 1 above)
Comment (efd6) [reply x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.yml:443]: That leaves the value in `checkpoint.bytes`.

File: x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.yml (hunk 2 above)
Comment (efd6) [reply x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.yml:484]: Same as above.

**Pull request merged**